src/conversions/std/ipaddr.rs

```diff
@@ -0,0 +1,110 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use crate::exceptions::PyValueError;
+use crate::sync::GILOnceCell;
+use crate::types::PyType;
+use crate::{intern, FromPyObject, IntoPy, Py, PyAny, PyObject, PyResult, Python, ToPyObject};
+
+impl FromPyObject<'_> for IpAddr {
+    fn extract(obj: &PyAny) -> PyResult<Self> {
+        match obj.getattr(intern!(obj.py(), "packed")) {
```

Comment:
Actually, I suppose there is a good question about the width of the conversion here. I feel that normally we do some kind of type check on the way in, so this is a break from most of PyO3's convention?

Reply:
Is the motivation not to check due to performance?

Reply:
So not type-checking for exactly an ip address on the to-rust direction is consistent with what we've done for the recent `rust_decimal` integration, so I think this is fine?

Reply:
Yes, I think so too. But it was a point of contention in that review (of the `rust_decimal` integration) so I was uncomfortable just merging it without further discussion. As written before, I am personally completely fine with board `extract` impls (even parsing strings) as the user's Rust code controls the target types and the order in which they are applied.

Reply:
FWIW the entire `.packed` check is a performance optimization. Python's IP address classes internally store the address as an int. The fallback path (`obj.str()?.to_str()?.parse()`) would also work for `IPv4Address` and `IPv6Address` instances. I have no strong opinions here and would be happy to go with whatever you feel is most consistent. :)

From an API user perspective, the ability to pass in strings would be very nice as that matches existing Python APIs (`socket.socket` etc take strings, not IPvXAddress instances).

Reply:
Yep, I think what we're converging on is that these generous-on-input conversions are quite practical.

At some point I wonder if we should go back over all the conversions we've implemented, categorise them into whether they are "generous" or strict, and make sure we're consistent in that approach?

Reply:
> At some point I wonder if we should go back over all the conversions we've implemented, categorise them into whether they are "generous" or strict, and make sure we're consistent in that approach?

It does sound like a good idea to check them all for consistency. I just wonder when someone will the time to do that since there quite a few of them by now. Certainly worth making an issue to be resolve before 1.0.

Reply:
Created #3226